### PR TITLE
server: support concurrent RM pushes

### PR DESCRIPTION
**Pull request by miki-totefu**
this makes the server support multiple open invoices for data to be pushed, which makes sending much faster

### CODE REVIEW

File: server/internal/pgdb/db.go [hunk 1]
```diff
@@ -676,14 +732,23 @@ func (db *DB) checkDatabaseSanity(tx *sql.Tx) error {
...
 		for tableName := range partitions {
 			if err := db.checkBulkTablespace(tx, tableName); err != nil {
 				return err
 			}
 
-			idxName := fmt.Sprintf("%s_rendezvous_point_insert_time_key", tableName)
+			// Determine the correct index name expected for the given
+			// table.
+			var idxName string
+			if ip == 2 {
```
Comment (dajohi) [inline server/internal/pgdb/db.go:746]: Could this `if partitions == db.redeemedPushesPartitions` instead?

File: server/svrpayments.go [hunk 2]
```diff
@@ -359,3 +424,15 @@ func (z *ZKS) areSubsPaid(ctx context.Context, r *rpc.SubscribeRoutedMessages, s
 
 	return err
 }
+
+func (z *ZKS) cancelLNInvoice(hash []byte) error {
```
Comment (dajohi) [inline server/svrpayments.go:428]: pass a ctx here

File: server/session.go [hunk 3]
```diff
@@ -471,5 +473,28 @@ func (z *ZKS) runNewSession(ctx context.Context, conn net.Conn, kx *session.KX)
 		z.logConn.Debugf("handleSession offline: %v", err)
 	}
 
+	// Cancel any outstanding invoices the session had that have not yet
+	// been redeemed.
+	for hash := range sc.lnPushHashes {
```
Comment (dajohi) [inline server/session.go:478]: does this need sc.Lock/Unlock?

File: server/svrpayments.go (hunk 2 above)
Comment (matheusd) [reply server/svrpayments.go]: done

**Review by dajohi — APPROVED** [review]

**Pull request merged**